pkg/cmd/grafana-cli/commands/conflict_user_command.go

```diff
@@ -101,16 +220,184 @@ func generateConflictUsersFile(r *ConflictResolver) (*os.File, error) {
+	// and the final output of the ingest operation
+	r.ValidUsers = newConflicts
+	r.BuildConflictBlocks(newConflicts, fmt.Sprintf)
+	return nil
+}
+
+func (r *ConflictResolver) MergeConflictingUsers(ctx context.Context, ss *sqlstore.SQLStore) error {
+	for block, users := range r.Blocks {
+		if len(users) < 2 {
+			return fmt.Errorf("not enough users to perform merge, found %d for id %s, should be at least 2", len(users), block)
+		}
+		var intoUser user.User
+		var intoUserId int64
+		var fromUserIds []int64
+
+		// creating a session for each block of users
+		// we want to rollback incase something happens during update / delete
+		sess := ss.NewSession(ctx)
+		defer sess.Close()
+		err := sess.Begin()
+		if err != nil {
+			return fmt.Errorf("could not open a db session: %w", err)
+		}
+		for _, u := range users {
+			if u.Direction == "+" {
+				id, err := strconv.ParseInt(u.Id, 10, 64)
+				if err != nil {
+					return fmt.Errorf("could not convert id in +")
+				}
+				intoUserId = id
+			} else if u.Direction == "-" {
+				id, err := strconv.ParseInt(u.Id, 10, 64)
+				if err != nil {
+					return fmt.Errorf("could not convert id in -")
+				}
+				fromUserIds = append(fromUserIds, id)
+			}
+		}
+		if _, err := sess.ID(intoUserId).Where(sqlstore.NotServiceAccountFilter(ss)).Get(&intoUser); err != nil {
+			return fmt.Errorf("could not find intoUser: %w", err)
+		}
+
+		for _, fromUserId := range fromUserIds {
+			var fromUser user.User
+			exists, err := sess.ID(fromUserId).Where(sqlstore.NotServiceAccountFilter(ss)).Get(&fromUser)
+			if err != nil {
+				return fmt.Errorf("could not find fromUser: %w", err)
+			}
+			if !exists {
+				fmt.Printf("user with id %d does not exist, skipping\n", fromUserId)
+			}
+			// // delete the user
+			delErr := ss.DeleteUserInSession(ctx, sess, &models.DeleteUserCommand{UserId: fromUserId})
+			if delErr != nil {
+				return fmt.Errorf("error during deletion of user: %w", delErr)
+			}
+		}
+		commitErr := sess.Commit()
+		if commitErr != nil {
+			return fmt.Errorf("could not commit operation for useridentification %s: %w", block, commitErr)
+		}
+		updateMainCommand := &models.UpdateUserCommand{
+			UserId: intoUser.ID,
+			Login:  strings.ToLower(intoUser.Login),
+			Email:  strings.ToLower(intoUser.Email),
+		}
+		updateErr := ss.UpdateUser(ctx, updateMainCommand)
+		if updateErr != nil {
+			return fmt.Errorf("could not update user: %w", updateErr)
+		}
+	}
+	return nil
+}
+
+/*
+hej@test.com+hej@test.com
+all of the permissions, roles and ownership will be transferred to the user.
++ id: 1, email: hej@test.com, login: hej@test.com
+these user(s) will be deleted and their permissions transferred.
+- id: 2, email: HEJ@TEST.COM, login: HEJ@TEST.COM
+- id: 3, email: hej@TEST.com, login: hej@TEST.com
+*/
+func (r *ConflictResolver) showChanges() {
+	if len(r.ValidUsers) == 0 {
+		fmt.Printf("no changes will take place as we have no valid users.\n")
+		return
+	}
+
+	var fileString string
```

Comment:
It's not a performance critical part, but I would use a `strings.Builder` instead of a string.

What do you think?